pkg/api/v1beta3/types.go

```diff
@@ -934,7 +934,7 @@ type ServiceSpec struct {
 	Ports []ServicePort `json:"ports" description:"ports exposed by the service"`
 
 	// This service will route traffic to pods having labels matching this selector. If null, no endpoints will be automatically created. If empty, all pods will be selected.
-	Selector map[string]string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
+	Selector string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
```

Comment:
The idea of merging a selector feels sketchy to me.  I think of a selector
(in its current form) as a single atomic field - more like a string than a
list, and I don't find it too egregious to say that PATCHing a selector is
a full replacement of that field.

On Thu, Apr 23, 2015 at 6:09 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1beta3/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7053#discussion_r29017480>
> :
>
> > @@ -934,7 +934,7 @@ type ServiceSpec struct {
> >  	Ports []ServicePort `json:"ports" description:"ports exposed by the service"`
> >
> >  	// This service will route traffic to pods having labels matching this selector. If null, no endpoints will be automatically created. If empty, all pods will be selected.
> > -	Selector map[string]string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
> > +	Selector string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
>
> Detail: Since we don't have any non-string-valued labels yet, and even
> when we do string-valued labels will be the most common by far, I'd omit
> "string" from the field names (just value and values).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7053/files#r29017480>
> .
>